pkg/kds/service/server.go

```diff
@@ -59,6 +61,29 @@ func (g *GlobalKDSServiceServer) StreamClusters(stream mesh_proto.GlobalKDSServi
 	})
 }
 
+func (g *GlobalKDSServiceServer) HealthCheck(ctx context.Context, _ *mesh_proto.ZoneHealthCheckRequest) (*mesh_proto.ZoneHealthCheckResponse, error) {
+	zone, err := util.ClientIDFromIncomingCtx(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	insight := system.NewZoneInsightResource()
+	if err := g.resManager.Get(ctx, insight, core_store.GetByKey(zone, model.NoMesh)); err != nil {
+		return nil, err
+	}
+	if insight.Spec.HealthCheck == nil {
+		insight.Spec.HealthCheck = &system_proto.HealthCheck{}
+	}
+
+	insight.Spec.HealthCheck.Time = timestamppb.Now()
+
+	if err := g.resManager.Update(ctx, insight); err != nil {
+		return nil, err
+	}
```

Comment:
I'd use `Upsert` and `WithConflictRetry` opt, because we are updating zone insight from multiple places. I'm also not sure we should return error to the client if we cannot update zone insight

Reply:
good point